### PR TITLE
Add user model

### PR DESCRIPTION
**Pull request by Tiana1203**

### CODE REVIEW

**Comment by Tiana1203** [comment]
im nor sure about BaseRepository.UpdateAsync cause it will generate 2 query for db for remove and add. Correct me if im wrong.

File: API/API/DataAccessLayer/Models/BaseModel.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace API.DataAccessLayer.Models
+{
+    public class BaseModel
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Models/BaseModel.cs]: abstract

File: API/API/DataAccessLayer/Repositories/BaseRespository.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            await _context.AddAsync(newEntity, cancellationToken);
+            await _context.SaveChangesAsync(cancellationToken);
+            return newEntity;
+        }
+
+        public async Task<T> UpdateAsync(T newEntity, CancellationToken cancellationToken = default)
+        {
+            await DeleteAsync(newEntity.Id, cancellationToken);
+            var res = await CreateAsync(newEntity, cancellationToken);
+            return res;
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Repositories/BaseRespository.cs]: wrong implementation of update method

File: API/API/DataAccessLayer/ShopdbContext.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            : base(options)
+        {
+            Database.EnsureCreated();
+            _configuration = configuration;
+        }
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            var connectionString = _configuration.GetConnectionString("DefaultConnection");
+            optionsBuilder.UseMySql(connectionString, new MySqlServerVersion(new Version(8, 0, 35)));
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/ShopdbContext.cs]: move this to DI

Comment (SUPERKISA8BIIT) [reply API/API/DataAccessLayer/ShopdbContext.cs]: you don't need configuration on data layer

File: API/API/DataAccessLayer/Configurations/UserConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            builder.ToTable("tbl_users");
+
+            builder.HasKey(u => u.Id);
+
+            builder.HasIndex(u => u.Email).IsUnique();
+
+            builder.Property(u => u.Id)
+                .HasColumnName("id");
+
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Configurations/UserConfiguration.cs]: id property configurated twice

File: API/API/DataAccessLayer/Configurations/UserConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+namespace API.DataAccessLayer.Configurations
+{
+    public class UserConfiguration : IEntityTypeConfiguration<User>
+    {
+        public void Configure(EntityTypeBuilder<User> builder)
+        {
+            builder.ToTable("tbl_users");
+
+            builder.HasKey(u => u.Id);
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Configurations/UserConfiguration.cs]: id is key by default

File: API/API/DataAccessLayer/Configurations/UserConfiguration.cs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+using Microsoft.EntityFrameworkCore;
+using API.DataAccessLayer.Models;
+
+namespace API.DataAccessLayer.Configurations
+{
+    public class UserConfiguration : IEntityTypeConfiguration<User>
+    {
+        public void Configure(EntityTypeBuilder<User> builder)
+        {
+            builder.ToTable("tbl_users");
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Configurations/UserConfiguration.cs]: use names generated by ef

File: API/API/DataAccessLayer/Configurations/UserConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+            builder.Property(u => u.Id)
+                .HasColumnName("id");
+
+            builder.Property(u => u.PasswordHash)
+                .HasMaxLength(100)
+                .IsRequired()
+                .HasColumnName("password_hash");
+
+            builder.Property(u => u.Email)
```
Comment (SUPERKISA8BIIT) [inline API/API/DataAccessLayer/Configurations/UserConfiguration.cs:17]: Email property configurated twice

File: API/API/BusinessLogicLayer/DTO/User/UserAddDTO.cs [hunk 8]
```diff
@@ -0,0 +1,12 @@
+﻿using System.ComponentModel.DataAnnotations;
+
+namespace API.BusinessLogicLayer.DTO.User
+{
+    public class UserAddDTO
+    {
+        [Required]
```
Comment (SUPERKISA8BIIT) [inline API/API/BusinessLogicLayer/DTO/User/UserAddDTO.cs]: use fluent validation

**Review by SUPERKISA8BIIT — CHANGES_REQUESTED** [review]
checked

**Comment by SUPERKISA8BIIT** [comment]
> im nor sure about BaseRepository.UpdateAsync cause it will generate 2 query for db for remove and add. Correct me if im wrong.

It wrong, because this will change id of entity

**Pull request merged**